### PR TITLE
Add AndroidX compatibility

### PR DESCRIPTION
**Pull request by lights0123**
[Capacitor 2 is switching to use the new AndroidX library][cap2], but this library depends on `android.support`. Instead of doing something to switch between the versions, this PR simply replaces the `@NonNull` annotation with a lambda function.

[cap2]: https://github.com/ionic-team/capacitor/blob/capacitor.2.0-beta.0/CHANGELOG.md

### CODE REVIEW

**Comment by lights0123** [comment]
I didn't see the past PRs #20 and #22, but this PR should be compatible with both `android` and `androidx`.

File: android/capacitor-fcm/build.gradle [hunk 1]
```diff
@@ -28,6 +28,10 @@ android {
     lintOptions {
         abortOnError false
     }
+    compileOptions {
```
Comment (stewwan) [inline android/capacitor-fcm/build.gradle]: can you explain a little bit about this addition?

File: android/capacitor-fcm/src/main/res/layout/bridge_layout_main.xml [hunk 2]
```diff
@@ -1,15 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (stewwan) [inline android/capacitor-fcm/src/main/res/layout/bridge_layout_main.xml]: why are you removing the bridge layout?

File: android/capacitor-fcm/build.gradle (hunk 1 above)
Comment (lights0123) [reply android/capacitor-fcm/build.gradle]: By using Java 8, which is [supported on all Android devices][java8], it's possible to remove the `android` vs `androidx`-specific packages (`android.support.annotation.NonNull`), and use lambdas instead.

[java8]: https://developer.android.com/studio/write/java8-support

File: android/capacitor-fcm/src/main/res/layout/bridge_layout_main.xml (hunk 2 above)
Comment (lights0123) [reply android/capacitor-fcm/src/main/res/layout/bridge_layout_main.xml]: This depends on `android.support.design.widget.CoordinatorLayout`, and I don't know if there's a way to dynamically switch between `android` and `androidx`. Is this even used in the package?

**Comment by lights0123** [comment]
@stewwan it's on.

File: android/capacitor-fcm/build.gradle (hunk 1 above)
Comment (stewwan) [reply android/capacitor-fcm/build.gradle]: interesting

File: android/capacitor-fcm/src/main/res/layout/bridge_layout_main.xml (hunk 2 above)
Comment (stewwan) [reply android/capacitor-fcm/src/main/res/layout/bridge_layout_main.xml]: I was thinking on publishing a v2 of this plugin using the "androidx way" for that. 

i.e.

```xml
<?xml version="1.0" encoding="utf-8"?>
<androidx.coordinatorlayout.widget.CoordinatorLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    tools:context="com.getcapacitor.BridgeActivity"
    >

    <WebView
        android:id="@+id/webview"
        android:layout_width="fill_parent"
        android:layout_height="fill_parent" />

</androidx.coordinatorlayout.widget.CoordinatorLayout>

```

I got a capacitor-fcm v2 version working locally already. thoughts?

Comment (lights0123) [reply android/capacitor-fcm/src/main/res/layout/bridge_layout_main.xml]: That's definitely an option. However, is that even required? I was trying to keep it so the same version would be compatible with both versions of capacitor.

Comment (stewwan) [reply android/capacitor-fcm/src/main/res/layout/bridge_layout_main.xml]: makes sense. I'll check it tomorrow.

**Comment by stewwan** [comment]
published to 2.0.0

**Pull request merged**